syntax = "proto3";
package protos.cust_orchestrator;

import "google/protobuf/empty.proto";

// =============================
// üß† SuperIntelligent Orchestrator Service
// =============================
service CustOrchestratorService {
  rpc ProcessCustomerQuery (ProcessCustomerQueryRequest) returns (ProcessCustomerQueryResponse);
  rpc HealthCheck (google.protobuf.Empty) returns (HealthResponse);
  rpc GetServiceInfo (google.protobuf.Empty) returns (ServiceInfoResponse);
}

// =============================
// üì© Request Messages
// =============================
message ProcessCustomerQueryRequest {
  string message = 1;
  string tenant_id = 2;
  string session_id = 3;
  map<string, string> metadata = 4;
}

// =============================
// üì§ Response Messages
// =============================
message ProcessCustomerQueryResponse {
  string response = 1;
  string intent = 2;
  SuperIntelligentMetadata superintelligent_metadata = 3;
  string session_id = 4;
  string trace_id = 5;
}

message SuperIntelligentMetadata {
  int32 tier = 1;
  string route = 2;
  double confidence = 3;
  double cost_rp = 4;
  string processing_method = 5;
  repeated string faq_sources = 6;
}

// =============================
// ‚ù§Ô∏è HealthCheck & Service Info
// =============================
message HealthResponse {
  string status = 1;
  string timestamp = 2;
}

message ServiceInfoResponse {
  string service_name = 1;
  string version = 2;
  repeated string features = 3;
  string grpc_version = 4;
}
